test/cli/autocorrect-kwsplat/test.rb

```diff
@@ -1,29 +0,0 @@
-# typed: strict
-extend T::Sig
-
-sig {params(pos: Integer, x: Integer).void}
-def requires_x(*pos, x:)
-end
-
-sig {params(pos: Integer, x: String).void}
-def optional_x(*pos, x: '')
-end
-
-opts = T::Hash[Symbol, Integer].new
-string_keys = T::Hash[String, Integer].new
-nilble_opts = T::Hash[Symbol, T.nilable(Integer)].new
-x = T.let(:x, Symbol)
-
-requires_x(0, 1, 2, opts)
-
-requires_x(0, 1, 2, **opts)
-
-requires_x(0, 1, 2, **opts, x: 0)
-
-requires_x(0, 1, 2, x: 0, **opts)
-
-requires_x(0, 1, 2, **{x => 0})
-
-optional_x(0, 1, 2, **string_keys)
```

Comment:
With the legacy handling we would also error here like it's a kwarg issue and propose to auto-correct with `T.unsafe` as usual. 
However, the actual error here is a type mismatch, because the code is trying to inline hash with string keys into keyword arguments, but it works only with symbol keyed hashes.

Legacy handling will wrap it with `T.unsafe` actually hiding that error! In the new handling the actual error is properly reported (check test/testdata/infer/ruby3_keyword_args.rb)